integration/integration_test.go

```diff
@@ -457,16 +458,42 @@ func TestDns(t *testing.T) {
 	t.Parallel()
 
 	dnsCmd := &command{
-		name:           "StartDnsGadget",
-		cmd:            fmt.Sprintf("$KUBECTL_GADGET trace dns -n %s", ns),
-		expectedRegexp: fmt.Sprintf(`%s\s+test-pod\s+OUTGOING\s+A\s+microsoft.com`, ns),
-		startAndStop:   true,
+		name:         "StartDnsGadget",
+		cmd:          fmt.Sprintf("$KUBECTL_GADGET trace dns -n %s -o json", ns),
+		startAndStop: true,
+		expectedOutputFn: func(output string) error {
+			expectedEntries := []*dnsTypes.Event{
+				{
+					Event:   buildBaseEvent(ns),
+					PktType: "OUTGOING",
+					DNSName: "inspektor-gadget.io.",
+					QType:   "A",
+				},
+				{
+					Event:   buildBaseEvent(ns),
+					PktType: "OUTGOING",
+					DNSName: "inspektor-gadget.io.",
+					QType:   "AAAA",
+				},
+			}
+
+			// DNS gadget doesn't provide container data. Remove it.
+			for _, entry := range expectedEntries {
+				entry.Container = ""
+			}
+
+			normalize := func(e *dnsTypes.Event) {
+				e.Node = ""
+			}
+
+			return expectEntriesToMatch(output, normalize, expectedEntries...)
```

Comment:
Out of scope for this PR: I don't remember if we ready discussed this, but do you think it would make sense to create another helper function where **all** the generated events must match one of the entries in `expectedEntries`? Right now, we stop at the first match.

Reply:
I'm not sure I follow here. Do you mean that for each generated event there should be at least one match on `expectedEntries`?

Reply:
Maybe it's useful, I'm not sure. I think we'll understand it when implementing more tests.